bubblewrap.c

```diff
@@ -1494,6 +1494,12 @@ parse_args_recurse (int          *argcp,
           if (argv[1][0] == 0 || endptr[0] != 0 || the_fd < 0)
             die ("Invalid fd: %s", argv[1]);
 
+          /* Specifying --args multiple times doesn't work; this just pacifies
```

Comment:
The man page says it does, and that does seem to be true:

```
$ rmdir /tmp/a /tmp/b
$ printf '--dir\0/tmp/a\0' > a
$ printf '--dir\0/tmp/b\0' > b
$ bwrap --dev-bind / / --args 3 --args 4 ls /tmp 3<a 4<b
(output contains a and b)
```

So I think `opt_args_data` might need to become a linked list, or some other mechanism that keeps all of the arguments blobs referenced so that static analyzers know they are OK to "leak" once per process.

Reply:
Eeek.  This is why we have code review!  It looks like at least flatpak (the main user of `--args`) is only appending it once, and for some reason I convinced myself the code was only designed to run once, but you're clearly right.

Reply:
https://github.com/containers/bubblewrap/pull/428
and then will rework this to be a linked list or strv after that lands.